### PR TITLE
Autogen softmax_with_cross_entropy

### PR DESCRIPTION
**Pull request by heavyrain-lzy**
### PR types

Others
### PR changes

OPs
### Describe

Pcard-67001
自动生成cross_entropy_with_softmax (softmax_with_cross_entropy)算子的静态图op代码，并移除原始手写代码。对比mlu和npu的kernel，已经没有对应`Backprop`输出，本次修改去除该输出。

### CODE REVIEW

File: paddle/phi/api/yaml/op_version.yaml [hunk 1]
```diff
@@ -169,6 +169,14 @@
         - add_input : ShiftsTensor
           comment : The number of places by which the elements of the tensor are shifted.
 
+- op : softmax_with_cross_entropy
+  version :
+    - checkpoint : Add a new attribute [use_softmax]
+      action :
+        - add_attr : force_cpu
```
Comment (zyfncg) [inline paddle/phi/api/yaml/op_version.yaml]: 这里应该是`use_softmax`

**Review by YuanRisheng — DISMISSED** [review]

File: paddle/phi/api/yaml/op_version.yaml (hunk 1 above)
Comment (heavyrain-lzy) [reply paddle/phi/api/yaml/op_version.yaml]: Done.Thanks.

**Review by jiahy0825 — APPROVED** [review]
LGTM

**Review by zyfncg — APPROVED** [review]

**Review by qili93 — APPROVED** [review]
LGTM

**Review by YuanRisheng — APPROVED** [review]

File: python/paddle/nn/functional/loss.py [hunk 2]
```diff
@@ -2768,13 +2761,6 @@ def cross_entropy(
...
 
         outputs = {'Softmax': softmax, 'Loss': out}
-        if core.is_compiled_with_custom_device(
-            "npu"
-        ) or core.is_compiled_with_custom_device("mlu"):
-            backprop = helper.create_variable_for_type_inference(
-                dtype=input.dtype
-            )
-            outputs['Backprop'] = backprop
         helper.append_op(
```
Comment (qili93) [inline python/paddle/nn/functional/loss.py:2764]: 这段代码是老的fluid算子需要用到 backprop，在新的PHI算子库实现中不再需要使用

**Review by chenwhql — APPROVED** [review]
